examples/show_hids.py

```diff
@@ -3,8 +3,27 @@
 
 #
 """
-Show all HID devices information
+Show all HID devices' information
+
+Written By Rene Aguirre (https://github.com/rene-aguirre) ~2016
+Updated by Gabriel Staples (https://www.ElectricRCAircraftGuy.com) 2 Feb 2017 
+-added "User selection" and this_script_dir_path stuff to allow writing output to a file in the same
+ dir as this python script, instead of just to the sys.stdout command prompt 
 """
+
+#User selection:
+PRINT_TO_FILE = True #set to True to print to a file in the same directory as this script, or False to print to the sys.stdout command prompt 
```

Comment:
@ElectricRCAircraftGuy my assumption is that library users knows how to deal with the windows command line.

I understand that this is useful, as a matter of fact, most of the time I use the script I capture the output to a file.

So, two comments:
a. Editing the file to change the setting seems awkward. How about changing it to a --file argparse option?

b. This definitely needs better usability, example, the script can provide instructions for usage. For this I'd suggest using argparse so a standard --help is shown.

Would you like to take the challenge? :-)

- Add one mandatory "--filename" option
- I'm fine with removing the printing at all, but if have a good idea for an option name (e.g. "--display") add an optional "print to screen too" option.
- Make the module __doc__ the actual script description.

Let me know if you think this is too much, some code to get you started:

```
    # this goes on top of main, remember to import argparse at top
    parser = argparse.ArgumentParser(description=__doc__,
                                     formatter_class=argparse.RawDescriptionHelpFormatter)
    parser.add_argument('outfile', help="A required output file",
                        type = argparse.FileType('w'))
    args = parser.parse_args(sys.argv[1:])
    # remove codecs setup logic, just call show_hids with output = args.output
    hid.core.show_hids(output = args.output)
```